### PR TITLE
Added support of pkg-config

### PR DESCRIPTION
**Pull request by dileping**
Have implemented support for pkg-config

Using this library in Swift ([Markdown](https://github.com/crossroadlabs/Markdown)) and `pkg-config` is the only proper way (at least for now) to import system libraries to Swift Package Manager.

Anyways, it will do no harm ;)

### CODE REVIEW

**Comment by Orc** [comment]
This is wonderful;  thank you!

**Comment by dileping** [comment]
@Orc Thanks for merging it in so fast. I will appreciate a lot if you could issue a new release as well so I can push it to package managers (brew, etc.)

**Comment by Orc** [comment]
Is the pkgconfig entry just for Swift, or are there other environments that use it?  I may modify it for 2.2.2 so that you need to `configure.sh --with-pkgconfig` (or `--with-swift`) to include the /usr/local/lib/pkgconfig stuff in `make install`, because I want to minimally clutter my already cluttered build environment :-)

**Comment by dileping** [comment]
@Orc I have already implemented everything and it installs properly (checked on Linux and Mac). You don't have to make any changes and make the build system more cumbersome.

It's not for Swift. It actually is a portable unix way to describe packages. You can read more about it here: [pkg-config](https://en.wikipedia.org/wiki/Pkg-config). Swift just uses same approach.

Also, in most packages that provide it, it's provided by default. No options required.

As an example of world class library that uses this approach take a look here: [libuv](https://github.com/libuv/libuv)

Also, let me know if you have any other questions. I will try to help.

**Comment by Orc** [comment]
Well, the annoying thing is that unless you've bought into the whole open source toolchain it doesn't show up on macos.   My development machine (10.12) doesn't have it, nor does my old FreeBSD 4.8 backwards-compatability machine.

Maybe I'll just tweak configure.sh so that it will look for pkg-config and generate the .pc file if pkg-config already exists?

**Comment by dileping** [comment]
@Orc The thing is that you don't need `pkg-config` to be installed to install the `.pc` file. It's generated with `automake`.

On the other side Swift Package Manager uses it directly. So it should be there even for Swift to work properly. Other tools may use the same file directly too.

Just to sum up. The file is of a config kind and makes no harm. Has no dependencies. Is it really an issue to leave it as is and install along with the library?

**Comment by Orc** [comment]
Automake is also another thing that doesn't come with macos or older versions of FreeBSD,  and personally I *really* don't want to have libraries dropping configuration files for that piece of gnuware over my systems.   Are there any machines out there that have automake, don't have pkg-config, but which use those .pc files as part of their build process?

**Comment by dileping** [comment]
@Orc 

Well, your `configure.sh` uses autoconf (which is a part of automake). I literally mean it does no harm and installs to the standard location (or does not if you don't do `make install`). In either way `make install` is a dirty way to install libraries nowadays. Still, `make install` is used internally by package managers that automatically determine which files were written.

1. The approach is not a *gnuware*, but rather is a portable *NIX approach.
2. Most world-class libraries provide it.
3. There is nothing additional required to build/use the library compared to the previous version. **No dependencies have to be installed.**
4. `libmarkdown.pc` file is generated from `libmarkdown.pc.in` with your standard procedure using `configure.sh`. **No dependencies are required to build the library.**
5. It can be used directly without installing `pkg-config` (it's a text file...)
6. Swift Package Manager (which is a standard for Swift in command line and provided by Apple) uses these files to import system wide installed libraries for both Mac and Linux. It's mandatory to have it to properly use the lib.
7. `.pc` files are the convenient portable way to describe a package for others to use it. I.e. what are the include paths, `-L` flags, etc.
8. I have never seen a library that *optionally* installs `.pc` file. It's always considered a benefit and increases the library portability. Obviously this becomes false if some installations omit the `.pc` file.

To sum up: you lose nothing, really. It just makes the library easier usable by others. I would love to use in in Swift, but unless it has `.pc` file it's kind of problematic. Consider it a *manifest* or a zero-conf *descriptor* of the library.

**Comment by Orc** [comment]
`configure.sh` doesn't actually used autoconf/automake/gnu configure.   It's a completely different  autoconfiguration implementation.

I think that the `--pkgconfig` option and/or autodetecting the presence of `pkg-config` is a good compromise between the collection of tools that wants to use .pc files and my desire to keep the installation as tidy as possible.

**Comment by dileping** [comment]
@Orc I understand you position. Can't agree... though, it's completely your choice what to do here.

Make it at least the default option, please, because package providers (apt, brew, etc.) might (and most probably will) miss the option.

I already had to fix the home-brew package for Discount, because it was built with wrong options and had multithreading issues.

So the least thing I'm asking about is to make it the default and if somebody (which I assume is quite a rare scenario) would think that he does not need it he could use `--without-pkgconfig`.

Just for convenience to make package makers (and package users) life easier.

Is this something you can agree with?

**Comment by dileping** [comment]
@Orc Thanks so much for releasing 2.2.1 with .pc file. Have already updated it for `brew`. See [PR](https://github.com/Homebrew/homebrew-core/pull/5205).

Many thanks, again!
